src/tree/values/nodes/StructuredNode.js

```diff
@@ -0,0 +1,92 @@
+import Node from "./Node";
+/**
+ * Represents a piece of structure that is present in the original text, but is not relevant for the further analysis
+ * of the text.
+ *
+ * Talking about HTML, this would encompass thing like <div>, <section>, <aside>, <fieldset> and other HTML block elements.
+ */
+class StructuredNode extends Node {
+	/**
+	 * Represents a piece of structure that is present in the original text, but is not relevant for the further
+	 * analysis of the text.
+	 *
+	 * Talking about HTML, this would encompass thing like <div>, <section>, <aside>, <fieldset> and other HTML block elements.
+	 *
+	 * @param {number}        startIndex  The index of the beginning of the structured node.
+	 * @param {number}        endIndex    The index of the end of the structured node.
+	 * @param {Array<Node>}   children    The sub-elements of the structured node.
+	 * @param {string}        tag         The tag used in the node.
+	 *
+	 * @returns {void}
+	 */
+	constructor( startIndex, endIndex, children, tag  ) {
+		super( "structuredNode", startIndex, endIndex );
+		this.children = children;
+		this.tag = tag;
+	}
+
+	/**
+	 * Clones this StructuredNode.
+	 *
+	 * @returns {Node} The cloned Node.
+	 */
+	clone() {
+		return new StructuredNode(
+			this.startIndex,
+			this.endIndex,
+			this.children,
+			this.tag
+		);
+	}
+
+	/**
+	 * Maps the given function to each Node in this tree.
+	 *
+	 * @param {mapCallback} mappingFunction 	The function that should be mapped to each Node in the tree.
+	 *
+	 * @returns {Node} A new tree, after the given function has been mapped on each Node.
+	 *
+	 * @abstract
+	 */
+	map( mappingFunction ) {
+		// Clone this node to avoid changing the original.
+		let clone = this.clone();
```

Comment:
I followed the same behavior as `Array.prototype.map`, since this also returns the changed array after mapping, without changing the original array in place.
I do agree that, in our case, changing the tree in-place would occur more often than not, since I expect we would have only one tree at the time.

Reply:
Either way, I removed `map` as well as `filter` for later implementation.